internal/client/client.go

```diff
@@ -0,0 +1,161 @@
+package client
+
+import (
+	"context"
+	"errors"
+	policiesv1 "github.com/kubewarden/kubewarden-controller/pkg/apis/policies/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"path/filepath"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	kubewardenPoliciesGroup   = "policies.kubewarden.io"
+	kubewardenPoliciesVersion = "v1"
+)
+
+// NamespacePolicies represents a map where the key is a namespace name, and the value are all policies that applies to this namespace
+type NamespacePolicies map[string][]policiesv1.Policy
+
+// A Client interacts with the kubernetes api
+type Client interface {
+	// GetPoliciesForANamespace gets all policies for a given namespace
+	GetPoliciesForANamespace(namespace string) ([]policiesv1.Policy, error)
+	// GetPoliciesForAllNamespaces gets all policies for all namespaces
+	GetPoliciesForAllNamespaces() ([]policiesv1.Policy, error)
+}
+
+type client struct {
+	k8sClient k8sClient.Client
+}
+
+// NewClient returns a client. It will try to use in-cluster config, which will work just if audit-scanner is deployed
+// inside a Pod. If in-cluster fails, it will try to fetch the kube config from the home dir. It will return an error
+// if both attempts fail.
+func NewClient() (Client, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return newClientFromHomeConfig()
+	}
+
+	return newClient(config)
+}
+
+func newClientFromHomeConfig() (Client, error) {
+	var kubeConfig string
+	if home := homedir.HomeDir(); home != "" {
+		kubeConfig = filepath.Join(home, ".kube", "config")
+	} else {
+		return nil, errors.New("can't get kubeconfig from home dir")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return newClient(config)
+}
+
+func newClient(config *rest.Config) (Client, error) {
+	// register Kubewarden policies in scheme
+	customScheme := scheme.Scheme
+	customScheme.AddKnownTypes(schema.GroupVersion{Group: kubewardenPoliciesGroup, Version: kubewardenPoliciesVersion}, &policiesv1.ClusterAdmissionPolicy{}, &policiesv1.AdmissionPolicy{}, &policiesv1.ClusterAdmissionPolicyList{}, &policiesv1.AdmissionPolicyList{})
```

Comment:
we don't need to register v1alpha2 policies, correct? Because they have been migrated?

Reply:
yes